### ISSUE TITLE
"Container Image is required" error when creating standalone init container

### ISSUE THREAD

**Issue by kravciak** [rancher/dashboard#2722]
If I create Init container without standard container, error message `Container Image is required` is misleading.
In UI container image filed has value: nginx:alpine.

**To reproduce:**
- Go to Cluster Explorer -> Deployments -> Create
- Fill in name and container (nginx / nginx:alpine)
- Select `Init Container`
- Create

**Current result is error message:**
```
"Container Image" is required
```

**Expected result:**
Error message saying I can't create init container without standard container.
When I create standalone init container from yaml kubectl gives this error:
```
ValidationError(Pod.spec): missing required field "containers" in io.k8s.api.core.v1.PodSpec
```

Tested on v2.5-head (31f00c1)

**Comment by kravciak** [comment]
Error message still does not make much sense, now it's `"Containers" is required`
![Screenshot from 2021-04-28 17-26-09](https://user-images.githubusercontent.com/538604/116430238-f534fe00-a846-11eb-9041-87b3159d61eb.png)

**Comment by mantis-toboggan-md** [comment]
This is in keeping with other UI-generated validation error messages for missing fields. Init containers don't appear in the containers pod spec field, so trying to create a deployment with only an initContainer should, indeed, tell you to define something in containers. Is this more helpful than the regular API error? Probably not, and there are other similarly meh error messages throughout the UI and other ways validation could be nicer, but that isn't going to be addressed in 2.5.8.

**Comment by kravciak** [comment]
Thank you, if that's expected behavior then I am closing this issue.
